File: apps/scheduler/src/repository/schemas/robot_game_match.py
from datetime import datetime
from typing import TypedDict, Literal
from bson import ObjectId


class RobotGameMatchParticipant(TypedDict):
    teamId: ObjectId | None
    tableId: ObjectId
    tableName: str
    queued: Literal[False]
    ready: Literal[False]
    present: Literal["no-show"]


class RobotGameMatch(TypedDict):
    divisionId: ObjectId
    round: int
    number: int
    stage: Literal["practice", "ranking"]
    called: Literal[False]
    status: Literal["not-started"]
    scheduledTime: datetime
    participants: list[RobotGameMatchParticipant]

File: libs/types/src/lib/old/schemas/robot-game-match.ts
import { ObjectId, WithId } from 'mongodb';
import { RobotGameMatchStage, RobotGameMatchStatus, RobotGameMatchPresent } from '../constants';
import { Team } from './team';

export interface RobotGameMatchBrief {
  divisionId: ObjectId;
  round: number;
  number: number;
  stage: RobotGameMatchStage;
  status: RobotGameMatchStatus;
  called: boolean;
  scheduledTime?: Date;
  startTime?: Date;
}

export interface RobotGameMatchParticipant {
  teamId: ObjectId | null;
  team?: WithId<Team>;
  tableId: ObjectId;
  tableName?: string;
  queued: boolean;
  present: RobotGameMatchPresent;
  ready: boolean;
}

export interface RobotGameMatch extends RobotGameMatchBrief {
  participants: Array<RobotGameMatchParticipant>;
}
